.asciidoctor/generate.sh

```diff
@@ -1,55 +1,77 @@
-                exit 0
-                ;;
-            -k|--keep)
-                shift
-                DELETE=1
-                shift
-                ;;
-            *)
-               echo "$1 is not a recognized flag!"
-               exit -1
-               ;;
-      esac
-done
+function readArguments() {
+    DELETE=0
 
-WORKING_DIR=${TRAVIS_BUILD_DIR:-${PWD}}
+    while test $# -gt 0; do
+           case "$1" in
+                -h|--help)
+                    echo "Builds documentation using asciidoc"
+                    echo "options:"
+                    echo "-h, --help                show brief help"
+                    echo "-k, --keep                keeps used container (useful for travis to export logs)"
+                    exit 0
+                    ;;
+                -k|--keep)
+                    shift
+                    DELETE=1
+                    shift
+                    ;;
+                *)
+                   echo "$1 is not a recognized flag!"
+                   exit -1
+                   ;;
+          esac
+    done
 
-TARGET_FOLDER=""
+    WORKING_DIR=${TRAVIS_BUILD_DIR:-${PWD}}
 
-if [ ! -z "${TRAVIS_PULL_REQUEST// }" ] && [ "${TRAVIS_PULL_REQUEST}" != "false" ]; then
-    TARGET_FOLDER="pr/${TRAVIS_PULL_REQUEST}"
-fi
+    TARGET_FOLDER=""
 
-if [ ! -z "${TRAVIS_TAG}" ]; then
-    TARGET_FOLDER="${TRAVIS_TAG}"
-fi
+    if [ ! -z "${TRAVIS_PULL_REQUEST// }" ] && [ "${TRAVIS_PULL_REQUEST}" != "false" ]; then
+        TARGET_FOLDER="pr/${TRAVIS_PULL_REQUEST}"
+    fi
 
-echo "Generating doc in $TARGET_FOLDER"
+    if [ ! -z "${TRAVIS_TAG}" ]; then
+        TARGET_FOLDER="${TRAVIS_TAG}"
+    fi
+}
 
-docker run -v ${WORKING_DIR}:/docs/:Z --name adoc-to-html rochdev/alpine-asciidoctor:mini asciidoctor \
-    -r /docs/.asciidoctor/lib/const-inline-macro.rb \
-    -r /docs/.asciidoctor/lib/copy-to-clipboard-inline-macro.rb \
-    -a generated-doc=true -a asciidoctor-source=/docs/docs \
-    /docs/README.adoc -o /docs/gh-pages/${TARGET_FOLDER}/index.html --trace;
+function generateDoc() {
+    echo "Generating doc from ${SOURCE} to ${TARGET}"
 
-docker ps -a | grep -q 'Exited (0)' && FAILED=0 || FAILED=1
+    docker run -v ${WORKING_DIR}:/docs/:Z -e PROJECT_ROOT=/docs/ --name adoc-to-html rochdev/alpine-asciidoctor:mini asciidoctor \
+        -r /docs/.asciidoctor/lib/const-inline-macro.rb \
+        -r /docs/.asciidoctor/lib/copy-to-clipboard-inline-macro.rb \
+        -a generated-doc=true -a asciidoctor-source=/docs/docs ${ASCIIDOC_ARGS} \
+        ${SOURCE} -o ${TARGET} --trace;
 
-if [ ${DELETE} -eq 0 ]; then
-    docker rm adoc-to-html;
-fi
+    docker ps -a | grep -q 'Exited (0)' && FAILED=0 || FAILED=1
+
+    if [ ${DELETE} -eq 0 ]; then
+        docker rm adoc-to-html;
+    fi
+
+    if [ ${FAILED} -eq 1 ]; then
+      echo "Failed generating the docs from ${SOURCE} to ${TARGET}"
+    fi
+}
+
+readArguments;
+SOURCE="/docs/README.adoc"
+TARGET="/docs/gh-pages/${TARGET_FOLDER}/index.html"
```

Comment:
If `TARGET_FOLDER` is an empty string, then it results in `//` in resulting TARGET, which doesn't do anything wrong. But maybe we can improve it further to avoid // in the resulting path of `TARGET`

Sample output:
`Generating doc from /docs/README.adoc to /docs/gh-pages//index.html`

Reply:
I can change it to:
```bash
if [[ -n ${TARGET_FOLDER} ]]; then
    TARGET_FOLDER="${TARGET_FOLDER}/"
fi
TARGET="/docs/gh-pages/${TARGET_FOLDER}index.html"
```
and the next usage to
```bash
TARGET="/docs/gh-pages/${TARGET_FOLDER}status/${FILE_PATH%.*}.html"
```
but, to be honest, the path without the missing string looks a bit weird to me. 
WDYT?

Reply:
I agree with you. we can merge this